jira_sync/sync_mgr.py

```diff
@@ -258,21 +258,27 @@ def reconcile_jira_forge_issues(self, matched_issues: Collection[MatchedIssue])
         """
         log.info("Reconciling matched JIRA and forge issues…")
         for jira_issue, forge_issue in matched_issues:
-            jira_assignee = (
+            forge_jira_assignee = (
                 forge_issue.repository.usermap.get(forge_issue.assignee)
                 if forge_issue.assignee
                 else None
             )
-            if jira_assignee != jira_issue.fields.assignee:
+            jira_assignee = jira_issue.fields.assignee
+            if (
+                jira_assignee
+                and forge_jira_assignee
+                and forge_jira_assignee not in (jira_assignee.key, jira_assignee.emailAddress)
+                or bool(forge_jira_assignee) is not bool(jira_assignee)
```

Comment:
Maybe this should be first check, so the rest could be skipped when this will be true.